File: scratch/update_reading_times.js
const fs = require("fs");
const path = require("path");

const jsonPath = path.resolve(__dirname, "../src/data/books.json");
const books = JSON.parse(fs.readFileSync(jsonPath, "utf8"));

const updatedBooks = books.map((book) => {
  const allText = Object.values(book.pages || {}).join(" ");
  const wordCount = allText.trim().split(/\s+/).length;
  const estimatedReadTime = Math.ceil(wordCount / 200);
  return {
    ...book,
    estimatedReadTime,
  };
});

fs.writeFileSync(jsonPath, JSON.stringify(updatedBooks, null, 2), "utf8");
console.log("Updated books.json with estimatedReadTime");

File: scratch/update_reading_times.py
import json
import os
import math

json_path = os.path.join(os.getcwd(), 'src/data/books.json')
with open(json_path, 'r', encoding='utf-8') as f:
    books = json.load(f)

for book in books:
    pages = book.get('pages', {})
    all_text = " ".join(pages.values())
    word_count = len(all_text.strip().split())
    # Avg reading speed: 200 words per minute
    estimated_read_time = math.ceil(word_count / 200)
    book['estimatedReadTime'] = estimated_read_time

with open(json_path, 'w', encoding='utf-8') as f:
    json.dump(books, f, ensure_ascii=False, indent=2)

print('Updated books.json with estimatedReadTime via Python')
